### PR TITLE
Cleanup on external API exposure

### LINKED ISSUE #924: VM exposes too much stuff
If one is doing an autocomplete on the `node` console there are just far too many instance members and functions being visible, and all in a very much inconsistent state regarding the naming of somewhat private methods with `_` or not. Here is an example screenshot on this:

Bildschirmfoto 2020-10-22 um 12 49 48

This is not very user friendly for people playing around with the VM and should be urgently cleaned up, optimally in a way that non-public members are not visible at all, minimally by adding underscores to give some orientation.

This should be done along the pre-release period since changes are pretty small with a significant effect, changes are also very much on the clean-up / consolidation side.

(also from a breaking/non-breaking aspect this is advisable to do now since people might use this exposed stuff unintentionally and will stumble if this is cleaned up along minor releases. If someone is missing something afterwards we can have an official discussion if something is worth to be re-exposed)

### PR DESCRIPTION
**Pull request by holgerd77**
Closes #924 

Ok, not the big sweeping blow I had hoped for but at least some substantial cleanup on what is exposed to the external API. Renamed some clearly internal members to underscore to better differentiate from the officially exposed functionality, also added `protected` keywords where possible so that these members are then not documented in the API docs.

Following reasoning and classifications for the respective member changes:

Categories

- 1 ^= Not meant to be public at all -> _ and private
- 2 ^= Not meant to be public but might be used in custom implementations -> _ and protected
- 3 ^= Should be part of the official API
- 4 ^= Not part of the official API, but is externally used -> _ and no explicit visibility

| Name | Category | Assignment | Change -> |
|---|---|---|---|
| opts | 2 | index.ts | _opts |
| allowUnlimitedContractSize | 2 | index.ts | _allowUnlimitedContractSize |
| isInitialized | 2 | index.ts | _isInitialized |
| _opcodes | 2 | index.ts | _opcodes, added protected |
| stateManager | 3 | index.ts | added comment for docs |
| blockchain | 3 (?) | index.ts | added comment for docs |
| _common | 4 | index.ts | heavily used in tests |
| _emit | 1 | index.ts | needs to be kept public due to code construction, added @hidden comment |
| _mcl | 1 | index.ts | needs to be kept public due to code construction, added @hidden comment |

I also fixed the docs generation accidentally including the tests along the way and regenerated documentation.

### CODE REVIEW

**Comment by holgerd77** [comment]
Note: `_emit` needs to be kept public as well, had this wrong in the first version of the initial comment (in case you read on mail), have updated.

File: packages/vm/lib/index.ts [hunk 1]
```diff
@@ -98,15 +98,35 @@ export interface VMOpts {
  * This class is an AsyncEventEmitter, please consult the README to learn how to use it.
  */
 export default class VM extends AsyncEventEmitter {
-  opts: VMOpts
-  _common: Common
+  
+  /**
+   * The StateManager used by the VM
+   */
   stateManager: StateManager
```
Comment (jochem-brouwer) [inline packages/vm/lib/index.ts:104]: Shouldn't `stateManager`, `blockchain`, 'common' be marked `readonly`?

Comment (holgerd77) [reply packages/vm/lib/index.ts]: Yes, good point, have added in e50af8003575ed7bb2b82b3a68de8a7e60354fd4. Realized that I actually had a wrong understanding of the [Readonly modifier](https://www.typescriptlang.org/docs/handbook/classes.html).

**Review by jochem-brouwer — APPROVED** [review]
Looks great! 👍

**Pull request merged**